test/config_parser_SUITE.erl

```diff
@@ -33,6 +33,21 @@ equivalence(Config) ->
     compare_unordered_lists(lists:filter(fun filter_config/1, Opts1), Opts2,
                             fun handle_config_option/2).
 
+miscellaneous(Config) ->
+    CfgPath = ejabberd_helper:data(Config, "miscellaneous.cfg"),
+    State1 = mongoose_config_parser_cfg:parse_file(CfgPath),
+    Hosts1 = mongoose_config_parser:state_to_host_opts(State1),
+    Opts1 = mongoose_config_parser:state_to_opts(State1),
+
+    TOMLPath = ejabberd_helper:data(Config, "miscellaneous.toml"),
+    State2 = mongoose_config_parser_toml:parse_file(TOMLPath),
+    Hosts2 = mongoose_config_parser:state_to_host_opts(State2),
+    Opts2 = mongoose_config_parser:state_to_opts(State2),
+
+    ?eq(Hosts1, Hosts2),
+    compare_unordered_lists(lists:filter(fun filter_config/1, Opts1), Opts2,
```

Comment:
Why do you need so many calls?
How about asserting them:
```suggestion
   ?assertEqual(lists:sort(Opts1), lists:sort(Opts2)).
```

Reply:
There are many corner cases where both config versions vary syntactically, but work the same. The complex functions are needed to handle them. And of course there can be nested unordered lists inside unordered lists.

Reply:
it's not quite that simple - there are nested structures in these lists and one call to `lists:sort` does not deal with all that